preproc/src/preproc.rs

```diff
@@ -35,24 +40,75 @@ impl Preprocessor for Pandocs {
     }
 
     fn run(&self, _: &PreprocessorContext, mut book: Book) -> Result<Book, Error> {
```

Comment:
Could we add some comments/documentation on what these new parts/functions of the preprocessor are doing?